src/main/java/com/powsybl/iidm/modification/topology/RemoveHvdcLine.java

```diff
@@ -0,0 +1,206 @@
+                .withDefaultMessage("Shunt compensator ${shuntId} has been ignored because it is not in the same voltage levels as the Lcc (${voltageLevelId1} or ${voltageLevelId2})")
+                .withValue("shuntId", shuntId)
+                .withValue("voltageLevelId1", voltageLevelId1)
+                .withValue("voltageLevelId2", voltageLevelId2)
+                .withSeverity(TypedValue.WARN_SEVERITY)
+                .build());
+    }
+
+    private static void notFoundShuntReport(Reporter reporter, String shuntId) {
+        reporter.report(Report.builder()
+                .withKey("notFoundShunt")
+                .withDefaultMessage("Shunt ${shuntId} not found")
+                .withValue("shuntId", shuntId)
+                .withSeverity(TypedValue.ERROR_SEVERITY)
+                .build());
+    }
+
+    @Override
+    public void apply(Network network, boolean throwException, ComputationManager computationManager, Reporter reporter) {
+        HvdcLine hvdcLine = network.getHvdcLine(hvdcLineId);
+        if (hvdcLine != null) {
+            HvdcConverterStation<?> hvdcConverterStation1 = hvdcLine.getConverterStation1();
+            HvdcConverterStation<?> hvdcConverterStation2 = hvdcLine.getConverterStation2();
+            Set<ShuntCompensator> shunts = null;
+            if (hvdcConverterStation1.getHvdcType() == HvdcConverterStation.HvdcType.LCC) { // in real-life cases, both converter stations are of the same type
+                shunts = shuntCompensatorIds.stream()
+                        .map(id -> getShuntCompensator(id, network, throwException, reporter))
+                        .filter(Objects::nonNull).collect(Collectors.toCollection(LinkedHashSet::new));
+            } else if (!shuntCompensatorIds.isEmpty()) { // VSC converter stations and defined shunts
+                String shuntIds = String.join(",", shuntCompensatorIds);
+                LOGGER.warn("Shunts {} are ignored since converter stations {} and {} are VSC", shuntIds, hvdcConverterStation1.getId(), hvdcConverterStation2.getId());
+                ignoredVscShunts(reporter, shuntIds, hvdcConverterStation1.getId(), hvdcConverterStation2.getId());
+            }
+            hvdcLine.remove();
+            removedHvdcLineReport(reporter, hvdcLineId);
+            LOGGER.info("Hvdc line {} has been removed", hvdcLineId);
+            // Remove the Shunt compensators that represent the filters of the LCC
+            removeShuntCompensators(network, hvdcConverterStation1, hvdcConverterStation2, shunts, throwException, computationManager, reporter);
+            removeConverterStations(network, hvdcConverterStation1, hvdcConverterStation2, throwException, computationManager, reporter);
+        } else {
+            LOGGER.error("Hvdc Line {} not found", hvdcLineId);
+            notFoundHvdcLineReport(reporter, hvdcLineId);
+            if (throwException) {
+                throw new PowsyblException("Hvdc Line " + hvdcLineId + " not found");
```

Comment:
Why `PowsyblException` ?
throw new NetworkModificationException(LOAD_NOT_FOUND, "Hvdc Line " + hvdcLineId + " not found");

Reply:
Sorry
Ignore this remark

Reply:
hack copy/paste from core